Mathlib/CategoryTheory/FiberedCategory/HomLift.lean

```diff
@@ -0,0 +1,248 @@
+  obtain rfl : S = T := by rw [← codomain_eq p f φ, domain_eq p (𝟙 T) ψ]
+  simpa using inferInstanceAs (p.IsHomLift (f ≫ 𝟙 S) (φ ≫ ψ))
+
+/-- If `φ : a ⟶ b` lifts `𝟙 T` and `ψ : b ⟶ c` lifts `f`, then `φ  ≫ ψ` lifts `f` -/
+lemma comp_lift_id_left {S T : 𝒮} {a b c : 𝒳} (f : S ⟶ T) (φ : a ⟶ b) (R : 𝒮)
+    [p.IsHomLift (𝟙 R) φ] (ψ : b ⟶ c) [p.IsHomLift f ψ] : p.IsHomLift f (φ ≫ ψ) := by
+  obtain rfl : R = S := by rw [← codomain_eq p (𝟙 R) φ, domain_eq p f ψ]
+  simpa using inferInstanceAs (p.IsHomLift (𝟙 R ≫ f) (φ ≫ ψ))
+
+lemma eqToHom_domain_lift_id {p : 𝒳 ⥤ 𝒮} {a b : 𝒳} (hab : a = b) {R : 𝒮} (hR : p.obj a = R) :
+    p.IsHomLift (𝟙 R) (eqToHom hab) := by
+  subst hR hab; simp
+
+lemma eqToHom_codomain_lift_id {p : 𝒳 ⥤ 𝒮} {a b : 𝒳} (hab : a = b) {S : 𝒮} (hS : p.obj b = S) :
+    p.IsHomLift (𝟙 S) (eqToHom hab) := by
+  subst hS hab; simp
+
+lemma id_lift_eqToHom_domain {p : 𝒳 ⥤ 𝒮} {R S : 𝒮} (hRS : R = S) {a : 𝒳} (ha : p.obj a = R) :
+    p.IsHomLift (eqToHom hRS) (𝟙 a) := by
+  subst hRS ha; simp
+
+lemma id_lift_eqToHom_codomain {p : 𝒳 ⥤ 𝒮} {R S : 𝒮} (hRS : R = S) {b : 𝒳} (hb : p.obj b = S) :
+    p.IsHomLift (eqToHom hRS) (𝟙 b) := by
+  subst hRS hb; simp
+
+instance comp_eqToHom_lift {R S : 𝒮} {a' a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : a' = a)
+    [p.IsHomLift f φ] : p.IsHomLift f (eqToHom h ≫ φ) := by
+  subst h; aesop
+
+instance eqToHom_comp_lift {R S : 𝒮} {a b b' : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : b = b')
+    [p.IsHomLift f φ] : p.IsHomLift f (φ ≫ eqToHom h) := by
+  subst h; aesop
+
+instance lift_eqToHom_comp {R' R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : R' = R)
+    [p.IsHomLift f φ] : p.IsHomLift (eqToHom h ≫ f) φ := by
+  subst h; aesop
+
+instance lift_comp_eqToHom {R S S': 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : S = S')
+    [p.IsHomLift f φ] : p.IsHomLift (f ≫ eqToHom h) φ := by
+  subst h; aesop
+
+@[simp]
+lemma comp_eqToHom_lift_iff {R S : 𝒮} {a' a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : a' = a) :
+    p.IsHomLift f (eqToHom h ≫ φ) ↔ p.IsHomLift f φ where
+  mp := by intro hφ'; subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma eqToHom_comp_lift_iff {R S : 𝒮} {a b b' : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : b = b') :
+    p.IsHomLift f (φ ≫ eqToHom h) ↔ p.IsHomLift f φ where
+  mp := by intro hφ'; subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma lift_eqToHom_comp_iff {R' R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : R' = R) :
+    p.IsHomLift (eqToHom h ≫ f) φ ↔ p.IsHomLift f φ where
+  mp := by intro hφ'; subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma lift_comp_eqToHom_iff {R S S' : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : S = S') :
+    p.IsHomLift (f ≫ eqToHom h) φ ↔ p.IsHomLift f φ where
+  mp := by intro hφ'; subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+/-- The isomorphism `R ≅ S` obtained from an isomorphism `φ : a ≅ b` lifting `f` -/
+def isoOfIsoLift  {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ≅ b) [p.IsHomLift f φ.hom] :
```

Comment:
The following syntax would automatically generate the next lemma:
```suggestion
@[simps hom]
def isoOfIsoLift  {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ≅ b) [p.IsHomLift f φ.hom] :
```

Reply:
Perfect! Thanks :)